src/lint/collect-spelling-diagnostics.ts

```diff
@@ -65,6 +65,10 @@ let matchers = [
     message:
       'clauses should be referenced using <emu-xref> tags rather than hardcoding clause numbers',
   },
+  {
+    pattern: /(?<=[^ \n]) +<\/p>/gu,
+    message: '<p> tags should not contain trailing whitespace',
```

Comment:
Nor leading whitespace, right? And why is `<p>` special here?

Reply:
I'll add leading whitespace as well.

`<p>` is special because it's the tag where this is the most likely to come up. Some tags like `<td>` intentionally contain trailing whitespace, so I don't want to do this for everything, and I don't want to come up with a full list of exactly which tags should and should not allow whitespace. If you have an uncontroversial version of such a list on hand, I'm happy to adopt it.

Reply:
I suppose I'd be fine with matching the list I used for the multiple consecutive spaces rule, i.e. forbidding leading / trailing spaces in everything except `<td>` and `<th>`, if you'd prefer that.